sale_require_po_doc/models/sale_order.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    customer_need_po = fields.Boolean(
+        related="partner_id.customer_need_po",
+        string='Customer Requires PO',
+    )
+    sale_doc = fields.Text(
+        related="partner_id.sale_doc",
+        string='Sales Require Documentation',
+    )
+    sale_document_option = fields.Selection(
+        [('no_need', 'No documentation needed for this Sale'),
+         ('required', 'Documentation required obtained and archived')],
+        string="Sale Documentation")
+    sale_document_note = fields.Text(
+        string='Sale Documentation Notes',
+    )
+
+    @api.multi
+    def action_confirm(self):
+        res = super(SaleOrder, self).action_confirm()
+        for order in self:
+            if order.customer_need_po and not order.client_order_ref:
+                raise ValidationError(
+                    _("You can not confirm sale order without \
+                        Customer reference."))
+            if order.sale_doc and not order.sale_document_option:
+                raise ValidationError(
+                    _("You can not confirm sale order without \
+                        Sale Documentation."))
+        return res
```

Comment:
Less resources spend if we do the validation upfront
```suggestion
        for order in self:
            if order.customer_need_po and not order.client_order_ref:
                raise ValidationError(
                    _("You can not confirm sale order without \
                        Customer reference."))
            if order.sale_doc and not order.sale_document_option:
                raise ValidationError(
                    _("You can not confirm sale order without \
                        Sale Documentation."))
        return super(SaleOrder, self).action_confirm()
```